player.py

```diff
@@ -9,25 +9,20 @@ def __init__(self):
         self.demand=[]
         self.bill = np.zeros(48) # prix de vente de l'électricité
         self.load= np.zeros(48) # chargement de la batterie (li)
+        self.penalty=np.zeros(48)
```

Comment:
I feel like this new attribute is not being used... is there something missing ?

Reply:
It is used by the manager in the new function compute_penalty in simulate.py. Moreover, we save this penalty at the end of each simulation so it is necessary.